### ISSUE TITLE
Monte Carlo Word Mover Distance 

### ISSUE THREAD

**Issue by robotcator** [RaRe-Technologies/gensim#1257]
This [repo](https://github.com/jwbaum91/MCWMD) have an implementation of Monte Carlo Word Movers Distance in Python with TensorFlow. Would it be integrated as a new feature in gensim?

**Comment by gojomo** [comment]
What kind of tasks does it work well on? Is there a paper/evaluation highlighting its advantages?

**Comment by robotcator** [comment]
@gojomo No, I just found it on a post and didn't find any paper about it.

**Comment by tmylk** [comment]
It would be a very nice addition if it works faster than current Gensim's WMD

**Comment by piskvorky** [comment]
@menshikh-iv this should be a nice Incubator project (the current WMD is somewhat hard to find and use -- Google doesn't even return Gensim at all!).
